commit ce55ce44c9c0cedf34a3b63c8bc077a94b255722 (refs/changes/08/3208/3)
Author: Pedro Palazon Candel <pedro@joyent.com>
Date:   2018-01-19T18:53:50+01:00 (1 year, 9 months ago)
    
    HEAD-2387 Add "dns_domain" to sapi service metadata
    Reviewed by: Josh Wilsdon <josh@wilsdon.ca>
    Approved by: Josh Wilsdon <josh@wilsdon.ca>

diff --git a/scripts/build-payload.js b/scripts/build-payload.js
index bd49f910..fc9fe1ca 100755
--- a/scripts/build-payload.js
+++ b/scripts/build-payload.js
@@ -6,7 +6,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /**
@@ -314,6 +314,7 @@ async.series([
         if (zone !== 'sapi') {
             return cb();
         }
+        obj.customer_metadata['dns_domain'] = config['dns_domain'];
 
         // Explicitly put this first sapi in proto mode.
         obj.customer_metadata['SAPI_PROTO_MODE'] = true;
diff --git a/scripts/sdc-init.js b/scripts/sdc-init.js
index d4bf00b9..03dba9e0 100755
--- a/scripts/sdc-init.js
+++ b/scripts/sdc-init.js
@@ -6,7 +6,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 /*
@@ -473,6 +473,9 @@ function filterServices(serviceList, cb) {
             }
             extras.metadata['assets-ip'] = self.config.assets_admin_ip;
             extras.metadata['user-script'] = data.toString();
+            if (service === 'sapi' && self.config.dns_domain) {
+                extras.metadata['dns_domain'] = self.config.dns_domain;
+            }
 
             // There's no need to pass the service defintion to
             // getOrCreateService() below, so remove it.
